valhalla/baldr/datetime.h

```diff
@@ -212,6 +212,38 @@ bool is_conditional_active(const bool type,
                            const uint64_t current_time,
                            const date::time_zone* time_zone);
 
+/**
+ * Provides a string representation of a condition in format close to the
+ * [opening hours](https://wiki.openstreetmap.org/wiki/Key:opening_hours/specification)
+ * @param   type          type of restriction kYMD or kNthDow
+ * @param   begin_hrs     begin hours
+ * @param   begin_mins    begin minutes
+ * @param   end_hrs       end hours
+ * @param   end_mins      end minutes
+ * @param   dow           days of the week to apply this restriction
+ * @param   begin_week    only set for kNthDow.  which week in the month
+ * @param   begin_month   begin month
+ * @param   begin_day_dow if kNthDow, then which dow to start the restriction.
+ *                        if kYMD then it is the day of the month
+ * @param   end_week      only set for kNthDow.  which week in the month
+ * @param   end_month     end month
+ * @param   end_day_dow   if kNthDow, then which dow to end the restriction.
+ *                        if kYMD then it is the day of the month
+ * @return String with time range that represents the condition
+ */
+std::string conditional_to_str(const bool type,
```

Comment:
can we not just pass the TimeDomain object into this and let it do its thing internally, why do we need to hav all the fields separate, is there use case where we dont have a timedomain already parsed? i see you are using it for a test, maybe you can just do an equality check between the proto fields and the real TimeDomain fields in the test rather than to_stringing them

Reply:
> can we not just pass the `TimeDomain` object into this and let it do its thing internally

I just wanted to keep it in par with `is_conditional_active()`, that takes the condition as separate fields.

> maybe you can just do an equality check between the proto fields and the real TimeDomain fields

Actually, this is what I do in tests, `conditional_to_str()` only used for pretty print if test failed.  And I used it in tests only because it already exists for `/locate` json output.

Reply:
Addressed in 23799605a30ab0ac75d7e4f49f05e03f4b1b2860